runtime/interpreter/interpreter_statement.go

```diff
@@ -0,0 +1,403 @@
+			LocationRange: interpreter.locationRange(statement),
+		})
+	}
+
+	err := interpreter.onEventEmitted(interpreter, event, eventType)
+	if err != nil {
+		panic(err)
+	}
+
+	// NOTE: no result, so it does *not* act like a return-statement
+	return trampoline.Done{}
+}
+
+func (interpreter *Interpreter) VisitPragmaDeclaration(_ *ast.PragmaDeclaration) ast.Repr {
+	return trampoline.Done{}
+}
+
+// VisitVariableDeclaration first visits the declaration's value,
+// then declares the variable with the name bound to the value
+func (interpreter *Interpreter) VisitVariableDeclaration(declaration *ast.VariableDeclaration) ast.Repr {
+
+	targetType := interpreter.Program.Elaboration.VariableDeclarationTargetTypes[declaration]
+	valueType := interpreter.Program.Elaboration.VariableDeclarationValueTypes[declaration]
+	secondValueType := interpreter.Program.Elaboration.VariableDeclarationSecondValueTypes[declaration]
+
+	result := interpreter.visitPotentialStorageRemoval(declaration.Value)
+
+	valueCopy := interpreter.copyAndConvert(result.(Value), valueType, targetType)
+
+	interpreter.declareVariable(
+		declaration.Identifier.Identifier,
+		valueCopy,
+	)
+
+	if declaration.SecondValue == nil {
+		// NOTE: ignore result, so it does *not* act like a return-statement
+		return trampoline.Done{}
+	}
+
+	return interpreter.visitAssignment(
+		declaration.Transfer.Operation,
+		declaration.Value,
+		valueType,
+		declaration.SecondValue,
+		secondValueType,
+		declaration,
+	)
+}
+
+func (interpreter *Interpreter) VisitAssignmentStatement(assignment *ast.AssignmentStatement) ast.Repr {
+	targetType := interpreter.Program.Elaboration.AssignmentStatementTargetTypes[assignment]
+	valueType := interpreter.Program.Elaboration.AssignmentStatementValueTypes[assignment]
+
+	target := assignment.Target
+	value := assignment.Value
+
+	return interpreter.visitAssignment(
+		assignment.Transfer.Operation,
+		target, targetType,
+		value, valueType,
+		assignment,
+	)
+}
+
+func (interpreter *Interpreter) VisitSwapStatement(swap *ast.SwapStatement) ast.Repr {
+
+	leftType := interpreter.Program.Elaboration.SwapStatementLeftTypes[swap]
+	rightType := interpreter.Program.Elaboration.SwapStatementRightTypes[swap]
+
+	// Evaluate the left expression
+	leftGetterSetter := interpreter.assignmentGetterSetter(swap.Left)
+	leftValue := leftGetterSetter.get()
+	if interpreter.movingStorageIndexExpression(swap.Left) != nil {
+		leftGetterSetter.set(NilValue{})
+	}
+
+	// Evaluate the right expression
+	rightGetterSetter := interpreter.assignmentGetterSetter(swap.Right)
+	rightValue := rightGetterSetter.get()
+	if interpreter.movingStorageIndexExpression(swap.Right) != nil {
+		rightGetterSetter.set(NilValue{})
+	}
+
+	// Add right value to left target
+	// and left value to right target
+
+	rightValueCopy := interpreter.copyAndConvert(rightValue.(Value), rightType, leftType)
+	leftValueCopy := interpreter.copyAndConvert(leftValue.(Value), leftType, rightType)
```

Comment:
Could do the same.
```suggestion
	rightValueCopy := interpreter.copyAndConvert(rightValue, rightType, leftType)
	leftValueCopy := interpreter.copyAndConvert(leftValue, leftType, rightType)
```

Reply:
Good catch! 👍 

Also removed the other unnecessary casts to `Value` in 9663ee9947b549463d4817a888cfa595c8180b63